doc/src/guides/solving/solve-polynomial.md

```diff
@@ -95,19 +96,149 @@ there is no guarantee that they have closed-form solutions at all, as explained
+{func}`~.real_roots` calls {func}`~sympy.polys.rootoftools.RootOf`, so you can
+get the same results by iterating over the number of roots of your equation:
+
+```py
+>>> [RootOf(expression, n) for n in range(0,3)]
+[-2, -2, 3]
+```
+
+### Approximate Numeric Solution With Root Multiplicities
+
+#### `nroots`
+
+{func}`~.nroots` gives an approximate numerical approximation to the roots of a
+polynomial. This example demonstrates that it can include numerical noise, for
+example a (negligible) imaginary component in what should be a real root:
+
+```py
+>>> nroots(expression)
+[3.0, -2.0 - 4.18482169793536e-14*I, -2.0 + 4.55872552179222e-14*I]
+```
+
+{func}`~.nroots`  is analogous to NumPy's {external:func}`~numpy.roots`
+function. Usually the difference between these two is that {func}`~.nroots` is
+more accurate but slower.
+
+A major advantage of {func}`~.nroots` is that it can compute numerical
+approximations of the roots of any polynomial whose coefficients can be
+numerically evaluated with {func}`~sympy.core.evalf` (that is, they do not have
+free symbols). Contrarily, symbolic solutions may not be possible for
+higher-order (fifth or greater) polynomials as explained by the [Abel-Ruffini
+theorem](https://en.wikipedia.org/wiki/Abel%E2%80%93Ruffini_theorem). Even if
+closed-form solutions are available, they may have so many terms that they are
+not useful in practice. You may therefore want to use {func}`~.nroots` to find
+approximate numeric solutions even if closed-form symbolic solutions are
+available.
+
+{func}`~.nroots` can fail sometimes for polynomials that are numerically ill
+conditioned, for example [Wilkinson's
+polynomial](https://en.wikipedia.org/wiki/Wilkinson%27s_polynomial).
+
+## Complex Roots
+
+For complex roots, similar functions can be used, for example {func}`~.solve`:
+
+```py
+>>> from sympy import solve, roots, nroots, real_roots, expand, RootOf, CRootOf, Symbol
+>>> from sympy import Poly
+>>> from sympy.abc import x
+>>> expression_complex = (x**2+4)**2 * (x-3)
+>>> solve(expression_complex, x, dict=True)
+[{x: 3}, {x: -2*I}, {x: 2*I}]
+```
+
+If the constants are symbolic, you may need to specify their domain for SymPy to
+recognize that the solutions are not real. For example, specifying that $a$ is
+positive leads to imaginary roots:
+
+```py
+>>> a = Symbol("a", positive=True)
+>>> symbolic_complex = (x**2+a)**2 * (x-3)
+>>> solve(symbolic_complex, x, dict=True)
+[{x: 3}, {x: -I*sqrt(a)}, {x: I*sqrt(a)}]
+```
+
+{func}`~.roots` will also find imaginary or complex roots:
+
+```py
+>>> roots(expression_complex, x)
+{3: 1, -2*I: 2, 2*I: 2}
+```
+
+{func}`~sympy.polys.rootoftools.RootOf` will also return complex roots:
+
+```py
+>>> [RootOf(expression_complex, n) for n in range(0,3)]
+[3, -2*I, -2*I]
+```
+
+{func}`~.real_roots` will only return the real roots and give no indication that
+there are complex roots, so use it with caution if your equation could have
+complex roots:
```

Comment:
I don't think it makes sense to caution here. The point of `real_roots` is that you only care about the real roots.